### PR TITLE
Small doc fixes

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: docs/jax_array_migration.md [hunk 1]
```diff
@@ -91,9 +91,9 @@ sharding = jax.sharding.NamedSharding(mesh, P('x'))
 
 sharded_x = jax.device_put(x, sharding)
 
-# `mul_sharded_x` and `sin_sharded_x` are sharded. `jit` is able to operate over a 
+# `dot_sharded_x` and `sin_sharded_x` are sharded. `jit` is able to operate over a
 # sharded array without copying data to a single device.
-mul_sharded_x = sharded_x @ sharded_x.T
+dot_sharded_x = sharded_x @ sharded_x.T
```
Comment (nouiz) [inline docs/jax_array_migration.md:96]: I wasn't sure what the intention was. Maybe the correction should be to change `@` with `*`.
Dot will mostly show that even if comm is needed to perform the operation, the output stay sharded.
I think it is good to show that more complex case.

Comment (yashk2810) [reply docs/jax_array_migration.md:96]: `@` is matrix multiplication AFAIK.  Maybe matmul_x is more accurate?

Comment (nouiz) [reply docs/jax_array_migration.md]: I amended the commit.

**Comment by nouiz** [comment]
@froystig This trivial PR is ready to merge. Can you look at it a last time?

**Review by mattjj — APPROVED** [review]
Thank you!

**Pull request merged**